packages/builder/src/components/design/settings/controls/multiFieldSelectState.test.ts

```diff
@@ -0,0 +1,41 @@
+import { describe, expect, it } from "vitest"
+import { MultiFieldSelectionState } from "./multiFieldSelectState"
+
+describe("MultiFieldSelectionState", () => {
+  it("keeps selection when options disappear temporarily", () => {
```

Comment:
Is this even an expected behaviour? Should we not make sure the source keeps being valid?

Reply:
This is what I mean. Is the issue that there is some loading state that needs to be handled and render things when ready, instead of trying to keep an up-to-date state in memory?